sdk/identity/azure-identity/azure/identity/_internal/decorators.py

```diff
@@ -33,7 +33,7 @@ def wrapper(*args, **kwargs):
                     upn = json_dict.get("upn", "unavailableUpn")
                     log_string = (
                         "[Authenticated account] Client ID: {}. Tenant ID: {}. User Principal Name: {}. "
-                        "Object ID (user): {}".format(json_dict["appid"], json_dict["tid"], upn, json_dict["oid"])
+                        "Object ID (user): {}".format(json_dict.get("appid",""), json_dict.get("tid",""), upn, json_dict.get("oid",""))
```

Comment:
Instead of an empty string, let's set the default to `"<unavailable>"` for each of these.

Pylint in the CI is also complaining about exceeding the 120 character line limit. Might be cleaner to put these in variables above like with `upn`.

Reply:
Both AzureCliCredential and ManagedIdentityCredential. Somewhere I read something that a "v2" type credential doesn't use appid but a "v1" does.

Reply:
Here is the guts of the JWT that blew it up:

```
{
  "aud": "<GUID>",
  "iss": "https://login.microsoftonline.com/<GUID>/v2.0",
  "iat": 1729272492,
  "nbf": 1729272492,
  "exp": 1729277166,
  "aio": "STUFF",
  "azp": "GUID",
  "azpacr": "0",
  "name": "Cory The Demo King",
  "oid": "42ff03bb-3709-425a-a788-58278553ee71",
  "preferred_username": "joeblow@contoso.com",
  "rh": "zzkdkdkd.",
  "scp": "access_as_user administer",
  "sub": "fjfjfjfjf",
  "tid": "<GUID>",
  "uti": "<GUID>",
  "ver": "2.0"
}
```

Do you mind if I change it up a bit? For example maybe dump the entire dictionary (which might not be the most safe) or maybe a subset?

Also I assume you are targeting python versions that don't support those f"" string (eg: `f"Here is my {variable}"`)

Reply:
Thanks for the info, and yep, looks like you're right! The `appid` GUID is only present in v1.0 tokens.

In order to maintain cross-language consistency amongst all the Azure Identity SDKs, let's just keep the logged keys to what we currently have. I don't have the full context, but these specific keys were [chosen](https://github.com/Azure/azure-sdk-for-python/issues/23203) a while back as being the most useful without causing too much noise.

And feel free to update these to use f-strings! The minimum Python version we currently target is 3.8, so we should be good using them.

Reply:
Hey, @coryking. Do you still have time to iterate on this? If not, I can help revise.

Reply:
No, I kinda moved onto other things. I trust y'alls judgment :-)